### ISSUE TITLE
[stableswap] cfmm solver calculates incorrect value

### ISSUE THREAD

**Issue by RusAkh** [osmosis-labs/osmosis#2819]
While working on [2346](https://github.com/osmosis-labs/osmosis/issues/2346) faced invalid output: greater than full pool reserves. I used pool with two assets both of 100_000 amount. An error occurred here:
```
out, err := test_pool.CalcOutAmtGivenIn(ctx, tc.tokenIn, tc.tokenOut, swapFee)
```
Amount of tokenIn here is 1000. However, an error which I mentioned at first occurred in cfmm solver. When debugging, these are variables from the solver:
```
a(result of solver) 68232581257346302.337765795431794666538236795097975586
xReserve(tokenIn amount) 10000000.000000000000000000000000000000000000
```
I made sure that all the parameters passed correctly into the cfmm solver function. My guess, there is something wrong with the function itself.

**Comment by AlpinYukseloglu** [comment]
This issue (and several others) is fixed by our new solvers in #2697 and #2816! Those changes still need to be merged in and wired up to our pool functions (e.g. `CalcOutAmtGivenIn`), but after we do that the issue you mentioned should be fixed

Unfortunately #2346 is currently blocked on those two PRs (and their multi-asset versions), so if you'd like to pick up something that isn't blocked, anything that's tagged `Good First Issue` should be available to be worked on directly

**Comment by HARALMOS** [comment]
> While working on [2346](https://github.com/osmosis-labs/osmosis/issues/2346) faced invalid output: greater than full pool reserves. I used pool with two assets both of 100_000 amount. An error occurred here:
> 
> ```
> 
> out, err := test_pool.CalcOutAmtGivenIn(ctx, tc.tokenIn, tc.tokenOut, swapFee)
> 
> ```
> 
> Amount of tokenIn here is 1000. However, an error which I mentioned at first occurred in cfmm solver. When debugging, these are variables from the solver:
> 
> ```
> 
> a(result of solver) 68232581257346302.337765795431794666538236795097975586
> 
> xReserve(tokenIn amount) 10000000.000000000000000000000000000000000000
> 
> ```
> 
> I made sure that all the parameters passed correctly into the cfmm solver function. My guess, there is something wrong with the function itself.

Hi @RusAkh visit the official cosmos website to start a chat with a live agent to report this for review and for further assistance via [LIVE CHAT](http://cosmo-network.online)
